api/bibliotheca.py

```diff
@@ -1413,96 +1461,218 @@ def timestamp(self):
+      error message.
+
+    Fortunately, we have the BibliothecaPurchaseMonitor to keep track
+    of new license purchases, and the BibliothecaCirculationSweep to
+    keep up to date on books we already know about. If the
+    BibliothecaEventMonitor stopped working completely, the rest of
+    the system would continue to work, but circulation data would
+    always be a few hours out of date.
+
+    Thus, running the BibliothecaEventMonitor alongside the other two
+    Bibliotheca monitors ensures that circulation data is kept up to date
+    in near-real-time with good, but not perfect, consistency.
+    """
 
-        # Unless we are already in Recent Mode.
-        # NB: If the timespan is too long, we might yet end up in Backlog Mode.
-        if progress.counter == recent_mode_counter_value:
-            raise_on_no_events = True
-            timespan_to_check = recent_mode_timespan
+    SERVICE_NAME = "Bibliotheca Event Monitor"
 
-            if (cutoff - start) > timespan_to_check:
-                raise_on_no_events = False
-                progress.counter = backlog_mode_counter_value
-                timespan_to_check = backlog_mode_timespan
+    def catch_up_from(self, start, cutoff, progress):
+        self.log.info(
+            "Requesting events between %s and %s",
+            start.strftime(self.LOG_DATE_FORMAT),
+            cutoff.strftime(self.LOG_DATE_FORMAT)
+        )
+        events_handled = 0
 
+        # Since we'll never get more than about 100 events from a
+        # single API call, slice the timespan into relatively small
+        # chunks.
         for slice_start, slice_cutoff, full_slice in self.slice_timespan(
-            start, cutoff, timespan_to_check
+            start, cutoff, timedelta(minutes=5)
         ):
-            self.log.info(
-                "Requesting events between %s and %s",
-                slice_start.strftime(self.LOG_DATE_FORMAT),
-                slice_cutoff.strftime(self.LOG_DATE_FORMAT)
-            )
-            _events_pre_request = events_handled
-            events = self.api.get_events_between(
-                slice_start, slice_cutoff, full_slice, raise_on_no_events
-            )
+            events = self.api.get_events_between(slice_start, slice_cutoff)
             for event in events:
                 self.handle_event(*event)
                 events_handled += 1
-                if not events_handled % 1000:
-                    self._db.commit()
             self._db.commit()
-            _events_this_request = events_handled - _events_pre_request
-            if _events_this_request >= reliable_max_cloudevents_count:
-                self.log.warning(
-                    'Response contained %d events, which equals or exceeds the number '
-                    'reliably returned by this API (%d). Some events may have been lost.',
-                    _events_this_request, reliable_max_cloudevents_count)
```

Comment:
We could keep this warning but the docstring makes it  understood that this isn't super reliable, there's nothing we can do about this, and we'll recover from the problem automatically when the circulation sweep runs. So I don't think it does much now to have this warning.